### ISSUE TITLE
RP ID should be optional

### ISSUE THREAD

**Issue by netanel-utila** [go-webauthn/webauthn#165]
### Version

0.8.4

### Description

Hi,

It seems like not passing RP ID is [invalid](https://github.com/go-webauthn/webauthn/blob/master/webauthn/types.go#L104), but it should be [valid](https://www.w3.org/TR/webauthn-2/#rp-id). In case we don't set the RP ID in the options, it will use the current domain. This will also make it usable in localhost, for example.

> By default, the [RP ID](https://www.w3.org/TR/webauthn-2/#rp-id) for a WebAuthn operation is set to the caller’s [origin](https://html.spec.whatwg.org/multipage/webappapis.html#concept-settings-object-origin)'s [effective domain](https://html.spec.whatwg.org/multipage/origin.html#concept-origin-effective-domain).

### Reproduction

Try to not pass RP ID

**Comment by james-d-elliott** [comment]
We can definitely make this change since it's supported in the spec. I suspect it may not have been in level 1 or prior and it got missed when I took over the library.

**Comment by james-d-elliott** [comment]
I've linked a fix to this. Would you mind trying it and see if it works? Reading the WDL it should be omitted I believe when this is intended.

**Comment by netanel-utila** [comment]
Thanks for the quick response. We'll try it, and let you know. We suspect that there are more places need to be changed.

**Comment by james-d-elliott** [comment]
Yeah this is the exact reason I wanted it checked. I think this will be mostly sufficient but there may be a validation on the response which needs changing specifically [this](https://www.w3.org/TR/webauthn-2/#sctn-registering-a-new-credential) (see step 13).

**Comment by netanel-utila** [comment]
Hi, it's also fails here:

https://github.com/go-webauthn/webauthn/blob/feda47adf57e1fa02737ce8434a9317e9602e291/protocol/authenticator.go#L366

**Comment by james-d-elliott** [comment]
Yeah that's what I was expecting above. So question is where do you want to supply the RP-ID? We need to supply it to validate that portion there. It could be at config time or at the time of the flow with a functional option.
